### DISCUSSION TITLE
Not an issue but maybe interesting

### DISCUSSION THREAD

**Discussion by revelator** [open-watcom/open-watcom-v2#22] [Ideas]
http://sourceforge.net/projects/cbadvanced/files/Open%20Watcom/stlport-watcom-2.0.7z/download

My port of STLPORT for Open Watcom. 

32 bit windows target only "like the old one".  

I used static runtime libraries though so this version can be used with later versions of Open Watcom with some adjustments to the config header.

**Comment by revelator** [comment]
Hmm the new cmath functions wreak havoc with stlports math definitions but im testing a new build now where i cast to std instead of stlports __std_alias (evc uses same approach) so if it works ill upload a new build soon.

**Comment by revelator** [comment]
sadly not working, patches to work around this are welcome.

**Comment by revelator** [comment]
Hmm i think i got lucky, i used the same strategy when porting SGI STL to open watcom and it seems to work now. Running some tests first to make sure though.

**Comment by revelator** [comment]
Ok got closer than last time but im still running into a snag atleast when it comes to using FILE.

'stlpmtx_std::FILE' has not been declared as a member error.

not sure but something tells me std::FILE is not exported by stlport, i also tried with the using std::FILE and its the same error so it seems its not exported.

**Comment by revelator** [comment]
Heh well this is somewhat weird, i can now build wxwidgets with stlport no problem but i cannot bootstrap  Open Watcom with it because it seems some changes in the last pull changed something in  watcoms cpplib. SGI STL now also fails to build watcom.

**Comment by abm-1980** [comment]
If it fully supports the STLport, what configurations do I have to make so that the compiler can work perfectly? Which version of STLport can I use? Is it the latest release?

**Comment by revelator** [comment]
watcoms stl supports more functions now than when i last built stlport, 
so it would need some patching to get around those.
Also watcoms stl uses non standard exports, so its not even certain it can be done without rewriting a good part of stlport. 
Clangs libc++ might be the better answer if we one day want a complete stl, it also supports C++11 and OrangeC from ladsoft allready uses it, i might get some ideas from looking at his version.
